### PR TITLE
Add PolicyBuilder

### PR DESCRIPTION
**Pull request by mcsherrylabs**
Make slot a Long (not Int)
Add test for DISH policy

### CODE REVIEW

File: src/main/scala/iog/psg/cardano/experimental/cli/model/Policy.scala [hunk 1]
```diff
@@ -46,7 +32,7 @@ object Policy {
   sealed trait Script
   object Script {
     case class Signature(keyHash: KeyHash[_ <: KeyType]) extends Script
-    case class Bound(slot: Int, after: Boolean) extends Script
+    case class Bound(slot: Long, after: Boolean) extends Script
```
Comment (adamsmo) [inline src/main/scala/iog/psg/cardano/experimental/cli/model/Policy.scala:35]: Maybe go up to BigInteger?

Comment (mcsherrylabs) [reply src/main/scala/iog/psg/cardano/experimental/cli/model/Policy.scala:35]: The type is defined as DWORD64 so Long should be enough. ... 

https://input-output-hk.github.io/ouroboros-network/cardano-slotting/Cardano-Slotting-Slot.html#t:SlotNo

File: src/test/scala/iog/psg/cardano/experimental/cli/model/PolicyBuilderSpec.scala [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    p.scripts.toList shouldBe List(Signature(kh))
+    p.kind shouldBe Policy.Kind.All
+  }
+
+  it should "create a minimal policy (default kind ALL sigs required) with one key" in {
+    val kh = KeyHash[Verification]("somestring")
+    val p = PolicyBuilder().withSignatureOf(kh).build
+    p.scripts.toList shouldBe List(Signature(kh))
+    p.kind shouldBe Policy.Kind.All
+  }
```
Comment (adamsmo) [inline src/test/scala/iog/psg/cardano/experimental/cli/model/PolicyBuilderSpec.scala:40]: Is this code the same as in `create a minimal policy with one key` ?

**Review by adamsmo — APPROVED** [review]
looks good, but two tests looks like are testing the same thing

**Pull request merged**